stl/inc/random

```diff
@@ -311,36 +311,28 @@ _NODISCARD _Uint _Next_linear_congruential_value(_Uint _Prev) noexcept {
     // Divide by _Ax:
     //   _Cx <= numeric_limits<T>::max()
     //   && (_Mx - 1) <= (numeric_limits<T>::max() - _Cx) / _Ax
-    if
-        _CONSTEXPR_IF(_Ax == 0) { // degenerate case; avoid divide by 0
-            (void) _Prev;
-            return static_cast<_Uint>(_Cx); // relies on _Mx == 0 || _Cx <= _Mx, N4741 [rand.eng.lcong]/3
-        }
-    else if
-        _CONSTEXPR_IF(
-            _Mx == 0) { // N4762 [rand.eng.lcong]/2: "If the template parameter m is 0, the modulus m used throughout
-                        // this subclause [rand.eng.lcong] is numeric_limits<result_type>::max() plus 1."
-                        // That is: Just do the multiply and let normal unsigned modulo take care of it
-            return static_cast<_Uint>(static_cast<_Uint>(_Ax * _Prev) + _Cx);
-        }
-    else if
-        _CONSTEXPR_IF(
-            _Cx <= UINT_MAX
-            && static_cast<_Uint>(_Mx - 1)
-                   <= (UINT_MAX - _Cx) / _Ax) { // unsigned int is sufficient to store intermediate calculation
-            const auto _Mul =
-                static_cast<unsigned int>(_Prev) * static_cast<unsigned int>(_Ax) + static_cast<unsigned int>(_Cx);
-            return static_cast<_Uint>(_Mul % _Mx);
-        }
-    else if
-        _CONSTEXPR_IF(
-            _Cx <= ULLONG_MAX
-            && static_cast<_Uint>(_Mx - 1)
-                   <= (ULLONG_MAX - _Cx) / _Ax) { // unsigned long long is sufficient to store intermediate calculation
-            const auto _Mul = static_cast<unsigned long long>(_Prev) * _Ax + _Cx;
-            return static_cast<_Uint>(_Mul % _Mx);
-        }
-    else { // no intermediate integral type fits; fall back to multiprecision
+    if _CONSTEXPR_IF (_Ax == 0) { // degenerate case; avoid divide by 0
+        (void) _Prev;
+        return static_cast<_Uint>(_Cx); // relies on _Mx == 0 || _Cx <= _Mx, N4741 [rand.eng.lcong]/3
+    } else if _CONSTEXPR_IF (_Mx == 0) { // N4762 [rand.eng.lcong]/2: "If the template parameter m is 0, the modulus m
+                                         // used throughout this subclause [rand.eng.lcong] is
+                                         // numeric_limits<result_type>::max() plus 1." That is: Just do the multiply
+                                         // and let normal unsigned modulo take care of it
+        return static_cast<_Uint>(static_cast<_Uint>(_Ax * _Prev) + _Cx);
+    } else if _CONSTEXPR_IF (_Cx <= UINT_MAX
+                             && static_cast<_Uint>(_Mx - 1)
+                                    <= (UINT_MAX - _Cx)
+                                           / _Ax) { // unsigned int is sufficient to store intermediate calculation

```

Comment:
Now that `_CONSTEXPR_IF` isn't horribly damaging this formatting, it appears that this comment should be detached to improve readability.